dataset/src/main/scala/frameless/Job.scala

```diff
@@ -0,0 +1,31 @@
+package frameless
+
+import org.apache.spark.SparkContext
+
+sealed abstract class Job[T](implicit sc: SparkContext) { self =>
+  /** Runs a new Spark job. */
+  def run: T
```

Comment:
I wonder if instead we should modify run to be

```scala
def run(sc: SparkContext): T
```

to avoid implicit config passing.

Also as a matter of style, I believe type params starting with `A` tend to be preferred